From: Chen Jun <chenjun102@huawei.com>
Subject: lockdep: make PROVE_RAW_LOCK_NESTING depend on PREEMPT_RT
Date: Thu, 16 Jun 2022 02:29:11 +0000

CONFIG_PROVE_RAW_LOCK_NESTING is used to check two things.
1. calling spin_lock in hardirq.
2. raw_spinlock - spinlock nesting.
The error messages like below:
[   13.485711] [ BUG: Invalid wait context ]

However there is no problem on non-PREEMPT_RT kernel. The messages may be
confusing and disturbing.

Therefore, do not check raw_spinlock - spinlock nesting
on non-PREEMPT_RT kernel.

Link: https://lkml.kernel.org/r/20220616022911.35826-1-chenjun102@huawei.com
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/Kconfig.debug |    1 +
 1 file changed, 1 insertion(+)

--- a/lib/Kconfig.debug~lockdep-make-prove_raw_lock_nesting-depend-on-preempt_rt
+++ a/lib/Kconfig.debug
@@ -1270,6 +1270,7 @@ config PROVE_LOCKING
 config PROVE_RAW_LOCK_NESTING
 	bool "Enable raw_spinlock - spinlock nesting checks"
 	depends on PROVE_LOCKING
+	depends on PREEMPT_RT
 	default n
 	help
 	 Enable the raw_spinlock vs. spinlock nesting checks which ensure
_
